demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -64,45 +69,19 @@ void Cnn::Init(const std::string &model_path, Core & ie, const std::string & dev
     // --------------------------- Creating infer request ------------------------------------------------
     infer_request_ = executable_network.CreateInferRequest();
     // ---------------------------------------------------------------------------------------------------
-
-    is_initialized_ = true;
 }
 
 InferenceEngine::BlobMap Cnn::Infer(const cv::Mat &frame) {
     std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
-
-    /* Resize manually and copy data from the image to the input blob */
-    InferenceEngine::LockedMemory<void> inputMapped =
-        InferenceEngine::as<InferenceEngine::MemoryBlob>(infer_request_.GetBlob(input_name_))->wmap();
-    float* input_data = inputMapped.as<float *>();
-
     cv::Mat image;
     if (channels_ == 1) {
          cv::cvtColor(frame, image, cv::COLOR_BGR2GRAY);
     } else {
         image = frame.clone();
     }
 
-    image.convertTo(image, CV_32F);
-    cv::resize(image, image, input_size_);
-
-    int image_size = input_size_.area();
-
-    if (channels_ == 3) {
-        for (int pid = 0; pid < image_size; ++pid) {
-            for (int ch = 0; ch < channels_; ++ch) {
-                input_data[ch * image_size + pid] = image.at<cv::Vec3f>(pid)[ch];
-            }
-        }
-    } else if (channels_ == 1) {
-        for (int pid = 0; pid < image_size; ++pid) {
-            input_data[pid] = image.at<float>(pid);
-        }
-    }
-    // ---------------------------------------------------------------------------------------------------
-
-    // --------------------------- Doing inference -------------------------------------------------------
-    /* Running the request synchronously */
+    auto blob = infer_request_.GetBlob(input_name_);
+    matU8ToBlob<uint8_t>(image, blob);
     infer_request_.Infer();
     // ---------------------------------------------------------------------------------------------------
```

Comment:
```suggestion
```
It corresponds to `// --------------------------- Doing inference -------------------------------------------------------`